### PR TITLE
docs(samples): Created Initial Sample for Generative AI app builder - Search

### PR DESCRIPTION
**Pull request by holtskinner**
- The Updated Client Library has been published.
- The Test Project has the API/Search Engine configured.
- Snippet Prefix exposed in tool.

### CODE REVIEW

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (Shabirmean) [inline genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:54]: This must be `com.google.api.grpc.search(...`

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (Shabirmean) [inline genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:55]: This must be `getResultsList()`

File: genappbuilder/pom.xml
Comment (Shabirmean) [inline genappbuilder/pom.xml:43]: The latest libraries bom only imports: `0.11.0` of the discoveryengine client. So I suggest you add the version explicitly here and add a comment to remove it when the `libraries-bom` has updated the version.

`<version>0.12.0</version>`

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (Shabirmean) [inline genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:22]: The `SearchRequest` and `SearchRequest` classes are defined in the `proto-google-cloud-discoveryengine-v1beta` package. Thus, you have to add that in the pom.xml after the `gproto-google-cloud-discoveryengine-v1beta`

```
<dependency>
      <groupId>com.google.api.grpc</groupId>
      <artifactId>proto-google-cloud-discoveryengine-v1beta</artifactId>
      <version>0.12.0</version>
    </dependency>
```

File: genappbuilder/pom.xml
Comment (Shabirmean) [inline genappbuilder/pom.xml:8]: Update here as well to be `discoveryengine`

File: genappbuilder/pom.xml
Comment (Shabirmean) [inline genappbuilder/pom.xml:4]: The id should use the API/product name. So I suggest you use `com.example. discoveryengine `

File: genappbuilder/pom.xml
Comment (Shabirmean) [inline genappbuilder/pom.xml:1]: The folder name must resemble the product or the API. So please use `discoveryengine` instead of genappbuilder`

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (Shabirmean) [inline genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:19]: Region tag prefix must match the product/API name. Please use `discoveryengine_search`
Update the END tag as well 
Also update all place where the region tag is used

**Review by Shabirmean — COMMENTED** [review]
Thank you for contributing this sample. Please TAL at the comments!

**Review by Shabirmean — CHANGES_REQUESTED** [review]
Needs making sure the CI runs

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (holtskinner) [reply genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:19]: This is a special case... Generative AI App Builder is using the Discovery Engine API and it has a distinct region tag. https://devrel.corp.google.com/admin/products/649

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (holtskinner) [reply genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:22]: That's really strange, why does this client library need it, but others don't?

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (holtskinner) [reply genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:54]: Not sure I understand. Why would this not use the search client?

File: genappbuilder/pom.xml
Comment (holtskinner) [reply genappbuilder/pom.xml:1]: See previous comment

File: genappbuilder/pom.xml
Comment (holtskinner) [reply genappbuilder/pom.xml:4]: Previous comment about product naming

File: genappbuilder/pom.xml
Comment (holtskinner) [reply genappbuilder/pom.xml:8]: previous comment about naming

File: genappbuilder/pom.xml
Comment (holtskinner) [reply genappbuilder/pom.xml:43]: That's interesting. It also didn't work when I added the version explicitly.
Does libraries-bom have access to the 0.12.0 version? It was just published recently

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (holtskinner) [reply genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:55]: Good catch, I missed this

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (Shabirmean) [reply genappbuilder/src/main/java/genappbuilder/v1beta/Search.java]: Sorry. That is what I was pointing to. I might have copy pasted something else.

File: genappbuilder/pom.xml
Comment (Shabirmean) [reply genappbuilder/pom.xml:43]: libraries-bom will automatically include it in the next release. But we are not sure when the cloud sdk team will publish the next release. SO until then we can explicitly force to use `0.12.0` by adding the version tag

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (Shabirmean) [reply genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:22]: It's because of the use of the `SearchRequest` and `SearchResponse` classes which are part of this dependency

File: genappbuilder/pom.xml
Comment (Shabirmean) [reply genappbuilder/pom.xml:4]: Please leave the comments `unresolved` for the original commentor to see your updates - the commentor shall resolve it once they find it addressed 🙌🏼 

Wrt to the naming - I still think the name here should match the client API. We can have a subdirectory inside it called `genappbuilder`. Also the region tag prefix can match `genappbuilder` since we have it added in devrel services. But the sample directory and the package names must be the client API:

Toplevel directory: `discoveryengine`
Subdirectory - `genappbuilder`
Pom xml artifact details:

```
<groupId>com.example.discoveryengine</groupId>
 <artifactId>genappbuilder-snippets</artifactId>
```

Comment (Shabirmean) [reply genappbuilder/pom.xml:4]: CC: @averikitsch

File: genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (holtskinner) [reply genappbuilder/src/main/java/genappbuilder/v1beta/Search.java]: That is bizarre, but it seems to have fixed the issue!

**Review by Shabirmean — CHANGES_REQUESTED** [review]
This looks great. Thanks for contributing the sample and also fixing the import issues. I have left a few more comments to polish up the samples to be consistent with the other samples in this repository.

Also one additional comment:
- Thanks for moving the sample to a new folder called `discoveryengine`
- I am assuming there will be more samples around discovery engine, which are not `genappbuilder`
- Thus, can we add the contents of this sample inside a new subdirectory called `genappbuilder` 
- So the path would be `discoveryengine/genappbuilder`

Also we need to add CODEOWNERS for this sample:
- Can you update the following file with the right Product/DPE team that must own this sample? [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS)

**Review by Shabirmean — APPROVED** [review]

**Comment by Shabirmean** [comment]
@balajismaniam @anguillanneuf Can someone from the @GoogleCloudPlatform/dee-data-ai team TAL at this before I can merge :)

File: discoveryengine/genappbuilder/src/main/java/genappbuilder/v1beta/Search.java
Comment (chenlei1216) [inline discoveryengine/genappbuilder/src/main/java/genappbuilder/v1beta/Search.java:59]: Let's add collectionId as well?

Comment (holtskinner) [reply discoveryengine/genappbuilder/src/main/java/genappbuilder/v1beta/Search.java]: Done

**Review by chenlei1216 — APPROVED** [review]

**Pull request merged**
